docs/DevGuide/PerformanceGuidelines.md

```diff
@@ -0,0 +1,19 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# General Performance Guidelines {#general_perf_guide}
+
+Below are some general guidelines for achieving decent performance.
+
+- One good measurement is worth more than a million expert opinions. We have a
+  `Benchmark` executable that uses Google Benchmark so one can compare different
+  implementations and see how they perform. This executable is only available in
+  release builds.
```

Comment:
It might be good to hyperlink to a SpECTRE guide on how to build/use this executable or at least provide the path to the executable. However, I can't seem to find a SpECTRE guide on the `Benchmark` executable, myself.

Since this PR is mainly meant to just kick off this performance guide and I'm not even sure there is a currently-existing `Benchmark` guide to be easily hyperlinked, maybe this is better for a future PR. Just wanted to mention it.

Reply:
I don't think there's anything written. I agree it should be added, but I would prefer that be done in a separate PR (maybe even by someone other than me(?), I don't have a lot of bandwidth to spare right now)

Reply:
It's a good suggestion @macedo22, but documentation PRs have the tendency to get out of hand, so we try to merge them fast and ask people to make changes in follow-ups. I added the `documentation` label to the PR.

Reply:
Sounds good!